src/notifications/analytics.ts

```diff
@@ -29,3 +31,27 @@ export const trackChangedNotificationSettings = (
     action,
   });
 };
+
+export const trackPushNotificationPermissionStatus = (
+  status: PushNotificationPermissionStatus
+) => {
+  analytics.identify(undefined, { notificationsPermissionStatus: status });
+};
+
+type PushNotificationPermissionStatus = 'enabled' | 'disabled' | 'never asked';
+
+export const resolveAndTrackPushNotificationPermissionStatus = async () => {
+  const permissionStatus = await getPermissionStatus();
+  let statusToReport: PushNotificationPermissionStatus = 'never asked';
```

Comment:
in case of `NOT_DETERMINED` ?

Reply:
yes and as a default case in general we report never asked